core/src/main/java/hudson/model/RunMap.java

```diff
@@ -228,13 +221,7 @@ public R put(R r) {
     }
 
     @Override public R getById(String id) {
-        int n;
-        try {
-            n = Integer.parseInt(id);
-        } catch (NumberFormatException x) {
-            n = runIdMigrator.findNumber(id);
-        }
-        return getByNumber(n);
+        return getByNumber(Integer.parseInt(id));
```

Comment:
Better to still catch `NumberFormatException` and return `null`, just in case some code calls this method with a nonnumeric argument for some reason. (Would be nice to deprecate this method in favor of something taking an `int`, but that would be a bigger project.)

Reply:
create https://issues.jenkins.io/browse/JENKINS-75476 and accordingly noted in the code comment.